include/crocoddyl/core/activations/smooth-1norm.hpp

```diff
@@ -50,8 +50,12 @@ class ActivationModelSmooth1NormTpl : public ActivationModelAbstractTpl<_Scalar>
    * @param[in] nr   Dimension of the residual vector
    * @param[in] eps  Smoothing factor (default: 1.)
    */
-  explicit ActivationModelSmooth1NormTpl(const std::size_t& nr, const Scalar& eps = Scalar(1.))
-      : Base(nr), eps_(eps){};
+  explicit ActivationModelSmooth1NormTpl(const std::size_t& nr, const Scalar& eps = Scalar(1.)) : Base(nr), eps_(eps) {
+    if (eps <= Scalar(0.)) {
```

Comment:
If the user wants eps to be zero, he should be allowed to do that. It is a choice that should be left to the user, not to us

Reply:
If we accept `eps=0`, then we don't have a smooth function.
Indeed, it becomes the absolute function, which it is not useful neither (there is a jump in the derivatives at the origin).

This was the rationality of imposing `<=`. I would vote to keep it. Do you agree with it?

Reply:
The default is 1. So the user is _definitely_ going to use zero by mistake.
Now if he _wants_ to set it to zero, we should not babyfeed him that it is not the expected use of this function. The user has chosen this function, and he should be allowed to play with it.

Thinking clearly, if he wants to set it to a negative value, this is a conscious choice on his part. It should not be restricted from our side. I think we should remove this check completely.

Reply:
Ok, I have done the follows:
  1. Accept eps=0
  2. Print a warning message for eps=0 

To be honest our solvers cannot handle 2 (derivative base).
A warn message might be useful for users that are not aware of it.